### ISSUE TITLE
Should fail if undefined path

### ISSUE THREAD

**Issue by lepicekmichal** [strmbx/retroxios#4]
Shouldn't retroxios throw some error if I define path parametr in url, but not provided it?

Etc: 

```
@GET("articles/{id}")
async articles(): Promise<AxiosResponse<any>> { return nothing() }
```

Should fail in these scenarios:
- have not sent id parameter
- not decorated it (or decorated it with wrong name)
- not defined default value

**Comment by tnychn** [comment]
Great suggestions.

If I understand it correctly, this is already implemented as **retroxios** throws when there's a path decorator but no corresponding key of it is found in the endpoint (i.e. decorated with the wrong key or no corresponding [replacement block](https://github.com/strmbx/retroxios#path-manipulation) is defined in the endpoint).

https://github.com/strmbx/retroxios/blob/a42da692a50692eae4745f7dc768f5dff7a01382/lib/request.ts#L55-L61

However, I think it's ok to allow users to define replacement block without having a corresponding path decorator because in some cases users may want to use the raw `{id}` string as part of the endpoint instead of as a replacement block itself.
Therefore, if there's no corresponding path decorator found, the replacement block will not be processed and it will become a raw string as part of the endpoint (even if a default value is specified inside the replacement block).

**Comment by lepicekmichal** [comment]
Yeah, i thought that might be the case. However, from my point of view I would let it crash.
For the same reason you use typescript instead of javascript, so you have some check - type check, initialization check, etc.
Feels it should be the same here. And if user wants it to be part of string, then escaping comes to the rescue.

Not sure what original retrofit does, but I guess better to keep it consistent. So If it's the same, then can close.

**Comment by lepicekmichal** [comment]
I have thought of a non-breaking solution. Add it to the config. strict=true would make this situation fail.
Maybe some other situations too.
